pkg/asset/manifests/infrastructure.go

```diff
@@ -81,14 +81,24 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
 			// not namespaced
 		},
 		Status: configv1.InfrastructureStatus{
-			InfrastructureName:   clusterID.InfraID,
-			Platform:             platform,
+			InfrastructureName: clusterID.InfraID,
+			Platform:           platform,
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: platform,
+			},
 			APIServerURL:         getInternalAPIServerURL(installConfig.Config),
 			APIServerInternalURL: getInternalAPIServerURL(installConfig.Config),
 			EtcdDiscoveryDomain:  getEtcdDiscoveryDomain(installConfig.Config),
 		},
 	}
 
+	switch installConfig.Config.Platform.Name() {
```

Comment:
could we collapse the two switches for platforms in to one?

Reply:
> could we collapse the two switches for platforms in to one?

Done with c52831893 -> ecaf8bfbf.